samples/boards/up_squared/gpio_counter/README.rst

```diff
@@ -0,0 +1,97 @@
+.. _up_squared gpio_counter:
+
+UP Squared
```

Comment:
Actually the title here should be ``UP Squared GPIO counter`` (and adjust the underline to be the same length as the new title).